### ISSUE TITLE
Exception when using internal terminology server

### ISSUE THREAD

**Issue by oliveregger** [ahdis/matchbox#236]
Caused by: org.hl7.fhir.utilities.SimpleHTTPClient$HTTPResultException: Invalid HTTP response 404 from http://localhost:8080/matchboxv3/tx-reg/resolve?fhirVersion=4.0&url=http%3A%2F%2Fnpu-terminology.org&usage=validation&nocache=1717128410417 (null) (Response in /var/folders/l2/kxp46ty52lj3tpdqm59ntp640000gn/T/http-log/fhir-http-7.log)
        at org.hl7.fhir.utilities.SimpleHTTPClient$HTTPResult.checkThrowException(SimpleHTTPClient.java:110)
        ... 125 more
........40..........60.........java.io.IOException
        at org.hl7.fhir.utilities.SimpleHTTPClient$HTTPResult.checkThrowException(SimpleHTTPClient.java:111)
        at org.hl7.fhir.utilities.json.parser.JsonParser.fetch(JsonParser.java:678)
        at org.hl7.fhir.utilities.json.parser.JsonParser.parseObjectFromUrl(JsonParser.java:79)
        at org.hl7.fhir.r5.terminologies.client.TerminologyClientManager.decideWhichServer(TerminologyClientManager.java:267)
        at org.hl7.fhir.r5.terminologies.client.TerminologyClientManager.findServerForSystem(TerminologyClientManager.java:228)
        at org.hl7.fhir.r5.terminologies.client.TerminologyClientManager.chooseServer(TerminologyClientManager.java:134)
        at org.hl7.fhir.r5.context.BaseWorkerContext.validateCode(BaseWorkerContext.java:1364)
        at org.hl7.fhir.r5.context.BaseWorkerContext.validateCode(BaseWorkerContext.java:1261)

Validate DocumentReference against http://fhir.ch/ig/ch-elm/StructureDefinition/PublishDocumentReference|1.3.0Validate Bundle against http://fhir.ch/ig/ch-elm/StructureDefinition/ch-elm-document|1.3.0Validate Composition against http://fhir.ch/ig/ch-elm/StructureDefinition/ch-elm-composition|1.3.0..java.io.IOException
        at org.hl7.fhir.utilities.SimpleHTTPClient$HTTPResult.checkThrowException(SimpleHTTPClient.java:111)
        at org.hl7.fhir.utilities.json.parser.JsonParser.fetch(JsonParser.java:678)
        at org.hl7.fhir.utilities.json.parser.JsonParser.parseObjectFromUrl(JsonParser.java:79)
        at org.hl7.fhir.r5.terminologies.client.TerminologyClientManager.decideWhichServer(TerminologyClientManager.java:267)

**Comment by oliveregger** [comment]
Caused by: org.hl7.fhir.utilities.SimpleHTTPClient$HTTPResultException: Invalid HTTP response 404 from http://localhost:8080/matchboxv3/tx-reg/resolve?fhirVersion=4.0&url=http%3A%2F%2Fsnomed.info%2Fsct%7Chttp%3A%2F%2Fsnomed.info%2Fsct%2F900000000000207008&usage=validation&nocache=1718375446128 (null) (Response in /var/folders/l2/kxp46ty52lj3tpdqm59ntp640000gn/T/http-log/fhir-http-8.log)

**Comment by oliveregger** [comment]
i think we should respond for the internal terminolog server our server, the format is as following from tx.fhir.org

https://tx.fhir.org/tx-reg/resolve?fhirVersion=4.0&url=http%3A%2F%2Fsnomed.info%2Fsct%7Chttp%3A%2F%2Fsnomed.info%2Fsct%2F900000000000207008&usage=validation&nocache=1718375446128

```json
{
  "formatVersion" : "1",
  "registry-url" : "https://raw.githubusercontent.com/FHIR/ig-registry/master/tx-servers.json",
  "candidates" : [
    {
      "server-name" : "tx.fhir.org",
      "url" : "http://tx.fhir.org/r4",
      "open" : true
    },
    {
      "server-name" : "HL7 Australia Server",
      "url" : "https://tx.ontoserver.csiro.au/fhir",
      "open" : true
    }
  ]
}

```

**Comment by qligier** [comment]
Ah, that's new.

**Comment by oliveregger** [comment]
fixed in Release 3.8.5
